cmd/snap/cmd_auto_import_test.go

```diff
@@ -129,35 +122,6 @@ func (s *SnapSuite) TestAutoImportAssertsNotImportedFromLoop(c *C) {
 	c.Check(s.Stderr(), Equals, "")
 }
 
-func (s *SnapSuite) TestAutoImportCandidatesHappy(c *C) {
-	dirs := make([]string, 4)
-	args := make([]interface{}, len(dirs))
-	files := make([]string, len(dirs))
-	for i := range dirs {
-		dirs[i] = c.MkDir()
-		args[i] = dirs[i]
-		files[i] = filepath.Join(dirs[i], "auto-import.assert")
-		err := ioutil.WriteFile(files[i], nil, 0644)
-		c.Assert(err, IsNil)
-	}
-
-	mockMountInfoFmtWithLoop := `
-too short
-24 0 8:18 / %[1]s rw,relatime foo ext3 /dev/meep2 no,separator
-24 0 8:18 / %[2]s rw,relatime - ext3 /dev/meep2 rw,errors=remount-ro,data=ordered
-24 0 8:18 / %[3]s rw,relatime opt:1 - ext4 /dev/meep3 rw,errors=remount-ro,data=ordered
-24 0 8:18 / %[4]s rw,relatime opt:1 opt:2 - ext2 /dev/meep1 rw,errors=remount-ro,data=ordered
-`
-
-	content := fmt.Sprintf(mockMountInfoFmtWithLoop, args...)
-	restore := snap.MockMountInfoPath(makeMockMountInfo(c, content))
-	defer restore()
-
-	l, err := snap.AutoImportCandidates()
-	c.Check(err, IsNil)
-	c.Check(l, DeepEquals, files[1:])
-}
-
```

Comment:
Why are we dropping this test? Is it covered by TestAutoImportUC20CandidatesIgnoresSystemPartitions?

Reply:
yes, also the first 3 lines, the empty string, the too short line, and the missing separator (`-`) are all no longer usable/parsable by osutil.LoadMountInfo so it does not run at all. I think the idea behind this test was to ensure that we skipped/ignored mount info entries that are invalid, but TBH I don't know why we need to support that case given that mountinfo is coming from the kernel and I think it's incredibly unlikely that the format of this will break or be different enough that we are unable to parse it. Even if it did change in a breaking way, other parts of snapd in the overlord package would also fall apart horribly so I don't think it's worth keeping that robustness support at the cost of having two implementations of parsing the mount table

Reply:
Agreed, it's unlikely this format changes, and if it does, it'll probably end up being a new file under /proc or /sys